src/Compilers/VisualBasic/Portable/Syntax/SyntaxFactory.vb

```diff
@@ -532,5 +534,118 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return SyntaxEquivalence.AreEquivalent(oldList.Node, newList.Node, ignoreChildNode, topLevel:=False)
         End Function
 
+        ''' <summary>
+        ''' Determines if a submission contains an incomplete LINQ query.
+        '''
+        ''' If the last expression of a submission is a LINQ query then it is
+        ''' complete only if the submission ends with a empty new line.
+        ''' For the examples below the first code submission the LINQ query is
+        ''' not completed, the second it is.
+        ''' <code>
+        ''' Dim x = FROM {1,2,3}
+        ''' </code>
+        ''' <code>
+        ''' Dim x = FROM {1,2,3}
+        '''
+        ''' </code>
+        ''' <param name="token">Last expected token of the last statement in a submission.</param>
+        ''' <param name="statementNode">Top level statement which token is part of.</param>
+        ''' <param name="endOfFileToken">Token that marks the end of submission.</param>
+        ''' </summary>
+        Private Shared Function IsPartOfIncompleteLinqQuery(token As SyntaxToken, statementNode As SyntaxNode, endOfFileToken As SyntaxToken) As Boolean
+            ' Checking if the submission ends with a new line.
+            For Each leadingTrivia In endOfFileToken.LeadingTrivia
+                If leadingTrivia.IsKind(SyntaxKind.EndOfLineTrivia) Then
+                    Return False
```

Comment:
Will this incorrectly return false for the following submission?
```
Dim x = From a In args Select _

```

Even if it does, it may not matter since the caller appears to check for a trailing underscore.